src/System.Runtime/tests/System/Reflection/TypeDelegatorTests.netcoreapp.cs

```diff
@@ -1,40 +1,113 @@
 using System.Collections.Generic;
+using System.Linq;
 using Xunit;
 
 namespace System.Reflection.Tests
 {
     public class TypeDelegatorNetcoreTests
     {
-        private static IEnumerable<object[]> SZArrayOrNotTypes()
-        {
-            yield return new object[] { typeof(int[]), true };
-            yield return new object[] { typeof(string[]), true };
-            yield return new object[] { typeof(void), false };
-            yield return new object[] { typeof(int), false };
-            yield return new object[] { typeof(int[]).MakeByRefType(), false };
-            yield return new object[] { typeof(int[,]), false };
-            yield return new object[] { typeof(TypeInfoTests), false };
-            yield return new object[] { Array.CreateInstance(typeof(int), new[] { 2 }, new[] { -1 }).GetType(), false };
-            yield return new object[] { Array.CreateInstance(typeof(int), new[] { 2 }, new[] { 1 }).GetType(), false };
-            yield return new object[] { Array.CreateInstance(typeof(int), new[] { 2 }, new[] { 0 }).GetType(), true };
-            yield return new object[] { typeof(int[][]), true };
-            yield return new object[] { Type.GetType("System.Int32[]"), true };
-            yield return new object[] { Type.GetType("System.Int32[*]"), false };
-            yield return new object[] { Type.GetType("System.Int32"), false };
-            yield return new object[] { typeof(int).MakeArrayType(), true };
-            yield return new object[] { typeof(int).MakeArrayType(1), false };
-            yield return new object[] { typeof(int).MakeArrayType().MakeArrayType(), true };
-            yield return new object[] { typeof(int).MakeArrayType(2), false };
-            yield return new object[] { typeof(Outside<int>.Inside<string>), false };
-            yield return new object[] { typeof(Outside<int>.Inside<string>[]), true };
-            yield return new object[] { typeof(Outside<int>.Inside<string>[,]), false };
-            yield return new object[] { Array.CreateInstance(typeof(Outside<int>.Inside<string>), new[] { 2 }, new[] { -1 }).GetType(), false };
-        }
-
-        [Theory, MemberData(nameof(SZArrayOrNotTypes))]
-        public void IsSZArray(Type type, bool expected)
-        {
-            Assert.Equal(expected, new TypeDelegator(type).IsSZArray);
+        private static readonly IEnumerable<Type> NonArrayBaseTypes = new Type[]
+         {
+            typeof(int),
+            typeof(void),
+            typeof(int*),
+            typeof(Outside),
+            typeof(Outside<>),
+            typeof(Outside<>).GetTypeInfo().GenericTypeParameters[0],
+            Type.GetTypeFromCLSID(default(Guid)),
+            new object().GetType().GetType()
+         };
+
+        [Fact]
+        public void IsSZArray_FalseForNonArrayTypes()
+        {
+            foreach (Type type in NonArrayBaseTypes)
+            {
+                Assert.False(new TypeDelegator(type).IsSZArray);
```

Comment:
After thinking about it a bit, I think the right way to test TypeDelegator is for the test to subclass Type itself, pass it in, then make sure each call to TypeDelegate results in a mirroring call to the mock Type. A single test class that does that makes more sense than duplicating the hundreds of tests of Type all over.

I'd say don't worry about testing TypeDelegator on this PR (TypeDelegator isn't exactly a big bug factory...) If you like, you can try something like the above but it needn't be part of this PR.